core/PRP/Physics/plGenericPhysical.cpp

```diff
@@ -733,3 +735,39 @@ void plGenericPhysical::setTMDBuffer(size_t tmdSize, const unsigned char* tmdBuf
     fTMDBuffer = new unsigned char[fTMDSize];
     memcpy(fTMDBuffer, tmdBuffer, fTMDSize);
 }
+
+void plGenericPhysical::calcSphereBounds(size_t numPoints, const hsVector3* points) {
+    hsBounds3 bounds;
+    for (size_t i = 0; i < numPoints; ++i)
+        bounds += points[i];
+
+    hsVector3 distance = (bounds.getMaxs() - bounds.getMins()) * .5f;
+    fRadius = std::max(distance.X, std::max(distance.Y, distance.Z));
+    fOffset = bounds.updateCenter();
+}
+
+void plGenericPhysical::calcBoxBounds(size_t numPoints, const hsVector3* points) {
+    fIndices.clear();
+    fVerts.clear();
+
+    hsBounds3Ext bounds;
+    for (size_t i = 0; i < numPoints; ++i)
+        bounds += points[i];
+    bounds.unalign();
+
+    // PhysX and ODE can do true boxes
+    fDimensions = (bounds.getMaxs() - bounds.getMins()) * .5f;
+    fOffset = bounds.updateCenter();
+
+    // Havok wants actual geometry
+    fVerts.resize(8);
+    fVerts[0] = fVerts[1] = fVerts[2] = fVerts[4] = bounds.getCorner();
+    fVerts[1] += bounds.getAxis(0);
+    fVerts[2] += bounds.getAxis(1);
+    fVerts[4] += bounds.getAxis(2);
```

Comment:
This looks like the correct way to do it for unaligned BBs, based off CWE's hsBounds3Ext's GetCorners(). I'm going to add that to libhsplasma for sanity's sake and remove these hacks.